libs/ngxtension/to-observable-signal/src/to-observable-signal.ts

```diff
@@ -6,30 +6,40 @@ import {
 } from '@angular/core';
 import {
 	toObservable,
+	toSignal,
 	type ToObservableOptions,
 } from '@angular/core/rxjs-interop';
-import type { Observable } from 'rxjs';
+import type { Observable, Subscribable } from 'rxjs';
 
 export type ObservableSignal<T> = Signal<T> & Observable<T>;
 
 export function toObservableSignal<T>(
-	s: WritableSignal<T>,
+	s: WritableSignal<T> | Observable<T> | Subscribable<T>,
 	options?: ToObservableOptions,
 ): WritableSignal<T> & Observable<T>;
 export function toObservableSignal<T>(
-	s: Signal<T>,
+	s: Signal<T> | Observable<T> | Subscribable<T>,
 	options?: ToObservableOptions,
 ): ObservableSignal<T>;
 
 export function toObservableSignal<T>(
-	s: Signal<T>,
+	source: Signal<T> | Observable<T> | Subscribable<T>,
 	options?: ToObservableOptions,
 ) {
 	if (isDevMode() && !options?.injector) {
 		assertInInjectionContext(toObservableSignal);
 	}
 
-	const obs = toObservable(s, options);
+	let s: Signal<T | undefined>;
+	let obs: Observable<T>;
+
+	if (typeof source === 'function') {
```

Comment:
suggestion: can we use `isSignal()` instead? This will probably remove the need of `as Signal<T>` down below